File: lib/models/proveedor.dart
class Proveedor {
  final int id;
  final String nombreProveedor;
  final String? telefono;
  final String? direccion;

  Proveedor({
    required this.id,
    required this.nombreProveedor,
    this.telefono,
    this.direccion,
  });

  factory Proveedor.fromJson(Map<String, dynamic> json) {
    return Proveedor(
      id: json['id'] as int,
      nombreProveedor: json['nombreProveedor'] as String,
      telefono: json['telefono'] as String?,
      direccion: json['direccion'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombreProveedor': nombreProveedor,
      'telefono': telefono,
      'direccion': direccion,
    };
  }

  @override
  String toString() {
    return 'Proveedor(id: $id, nombreProveedor: $nombreProveedor, telefono: $telefono, direccion: $direccion)';
  }
} 

File: Backend/SanJoseAPI/Models/Proveedor.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SanJoseAPI.Models
{
    [Table("proveedores")]
    public class Proveedor
    {
        [Key]
        [Column("id_proveedor")]
        public int Id { get; set; }

        [Required]
        [Column("nombre_proveedor")]
        [StringLength(100)]
        public string NombreProveedor { get; set; }

        [Column("telefono")]
        [StringLength(20)]
        public string Telefono { get; set; }

        [Column("direccion")]
        public string Direccion { get; set; }

        // Relaciones
        public ICollection<Insumo> Insumos { get; set; }
    }
} 